internal/services/oracle/oracle_exascale_db_storage_vault_resource.go

```diff
@@ -0,0 +1,331 @@
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.StorageVaultName,
+		},
+
+		"high_capacity_database_storage_input": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"total_size_in_gbs": {
+						Type:     pluginsdk.TypeInt,
+						Required: true,
+					},
+				},
+			},
+		},
+
+		// Optional
+		"high_capacity_database_storage": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"available_size_in_gbs": {
+						Type:     pluginsdk.TypeInt,
+						Optional: true,
+						Computed: true,
+					},
+					"total_size_in_gbs": {
+						Type:     pluginsdk.TypeInt,
+						Optional: true,
+						Computed: true,
+					},
+				},
+			},
+		},
+
+		"time_zone": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			Default:  "UTC",
+		},
+
+		"tags": commonschema.Tags(),
+
+		"zones": commonschema.ZonesMultipleRequiredForceNew(),
+	}
+}
+
+func (ExascaleDbStorageVaultResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (ExascaleDbStorageVaultResource) ModelObject() interface{} {
+	return &ExascaleDbStorageVaultResource{}
+}
+
+func (ExascaleDbStorageVaultResource) ResourceType() string {
+	return "azurerm_oracle_exascale_db_storage_vault"
+}
+
+func (r ExascaleDbStorageVaultResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 120 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient.ExascaleDbStorageVaults
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model ExascaleDbStorageVaultResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			id := exascaledbstoragevaults.NewExascaleDbStorageVaultID(subscriptionId,
+				model.ResourceGroupName,
+				model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			param := exascaledbstoragevaults.ExascaleDbStorageVault{
+				Name:     pointer.To(model.Name),
+				Location: location.Normalize(model.Location),
+				Zones:    pointer.To(model.Zones),
+				Tags:     pointer.To(model.Tags),
+				Properties: &exascaledbstoragevaults.ExascaleDbStorageVaultProperties{
+					DisplayName:                   model.DisplayName,
+					AdditionalFlashCacheInPercent: pointer.To(model.AdditionalFlashCacheInPercent),
+					Description:                   pointer.To(model.Description),
+					TimeZone:                      pointer.To(model.TimeZone),
+				},
+			}
+
+			if len(model.HighCapacityDatabaseStorageInput) > 0 {
```

Comment:
Since this is a required property, we don't need to do the check, instead we shall inline this property in its declaration above.